tests/src/Kernel/Entity/GetEntityGroupsTest.php

```diff
@@ -173,22 +174,67 @@ public function testOtherGroups() {
   }
 
   /**
+   * Tests member methods for states that other groups users are added to.
+   */
+  public function testIsMemberStates() {
+    // Add user to group 1 should now return that group only.
+    $membership = $this->createMembership($this->user3, $this->group1);
+
+    // Default param is ACTIVE.
+    $this->assertTrue(Og::isMember($this->group1, $this->user3));
+
+    $this->assertFalse(Og::isMember($this->group1, $this->user3, [OgMembershipInterface::STATE_PENDING]));
+    $this->assertFalse(Og::isMember($this->group1, $this->user3, [OgMembershipInterface::STATE_BLOCKED]));
+    $this->assertFalse(Og::isMember($this->group1, $this->user3, [OgMembershipInterface::STATE_PENDING, OgMembershipInterface::STATE_BLOCKED]));
+    $this->assertFalse(Og::isMemberPending($this->group1, $this->user3));
+    $this->assertFalse(Og::isMemberBlocked($this->group1, $this->user3));
+
+    // Change the membership state to PENDING.
+    $membership->setState(OgMembershipInterface::STATE_PENDING)->save();
+
+    Og::invalidateCache();
+
+    $this->assertTrue(Og::isMember($this->group1, $this->user3, [OgMembershipInterface::STATE_PENDING]));
+    $this->assertTrue(Og::isMemberPending($this->group1, $this->user3));
+    $this->assertTrue(Og::isMember($this->group1, $this->user3, [OgMembershipInterface::STATE_PENDING, OgMembershipInterface::STATE_BLOCKED]));
+
+    $this->assertFalse(Og::isMember($this->group1, $this->user3));
+    $this->assertFalse(Og::isMember($this->group1, $this->user3, [OgMembershipInterface::STATE_BLOCKED]));
+    $this->assertFalse(Og::isMemberBlocked($this->group1, $this->user3));
+
+    // Change the membership state to BLOCKED.
+    $membership->setState(OgMembershipInterface::STATE_BLOCKED)->save();
+
+    Og::invalidateCache();
```

Comment:
Yeah, hopefully we can fix this and invalidate our caches for reals. Def f/u material. Well done for offering :)